integration/swarm/leadership_test.go

```diff
@@ -0,0 +1,209 @@
+package swarm // import "github.com/docker/docker/integration/swarm"
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/integration/internal/swarm"
+	"github.com/docker/docker/internal/test/daemon"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+	"gotest.tools/poll"
+	"gotest.tools/skip"
+)
+
+// TestSwarmLeaderElection tests the swarm leader election behavior. It creates
+// three daemons in a swarm, all managers, and then covers three cases:
+//   1. Adding nodes after the first one should not cause a leader election
+//   2. Stopping the leader should cause a leader election
+//   3. Adding the stopped node back should not cause a leader election
+func TestSwarmLeaderElection(t *testing.T) {
+	defer setupTest(t)()
+
+	skip.If(t, runtime.GOARCH == "s390x", "disabled on s390x")
+	skip.If(t, runtime.GOARCH == "ppc64le", "disabled on ppc64le")
+
+	daemons := [3]*daemon.Daemon{}
```

Comment:
I chose an array over a slice because this test is written for exactly 3 daemons, no more, no less, If it wouldn't make iteration impossible, I likely would have outright used 3 separate variables. That is, I chose an array specifically because it is less flexible. I can change it if my thought process isn't quite right, though.